lib/miq_expression.rb

```diff
@@ -285,7 +285,7 @@ def self._to_ruby(exp, context_type, tz)
 
   def to_sql(tz = nil)
     tz ||= "UTC"
-    @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
+    @pexp, attrs = preprocess_for_sql(@exp.deep_clone.except!(:token))
```

Comment:
I'd prefer we don't go this route

Reply:
@kbrock is you objection because  it looks as a "dirty fix" ?
on the bright side it is very save (modification done on clone) and will not do any harm now and in the future

I am not clear what you mean by solution 3.
Everywhere in the code of ```MiqExpression``` we assumed that only single entry in the top level of Hash is present, this code ```operator = exp.keys.first``` is all over

Reply:
@kbrock, I'm looking at this with @yrudman. It seems like options 4 is the best, perhaps safe,  solution. But I'm not clear why it's your least preffered option. Perhaps the three of us can discus?